### PR TITLE
Various WASM AOT changes

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/System.Private.CoreLib/src/System/Array.Mono.cs [hunk 1]
```diff
@@ -144,6 +144,11 @@ private static void Copy(Array sourceArray, int sourceIndex, Array destinationAr
             if (FastCopy(sourceArray, sourceIndex, destinationArray, destinationIndex, length))
                 return;
 
+            CopySlow(sourceArray, sourceIndex, destinationArray, destinationIndex, length, reliable);
```
Comment (kasperk81) [inline src/mono/System.Private.CoreLib/src/System/Array.Mono.cs:147]: CopySlow -> SlowCopy or rename the other one to CopyFast?

**Review by lambdageek — APPROVED** [review]

**Review by imhameed — APPROVED** [review]

**Comment by vargaz** [comment]
Failures appear unrelated

**Pull request merged**